include/deal.II/matrix_free/cuda_hanging_nodes_internal.h

```diff
@@ -58,9 +58,11 @@ namespace CUDAWrappers
        */
       template <typename CellIterator>
       void
-      setup_constraints(std::vector<types::global_dof_index> &dof_indices,
-                        const CellIterator &                  cell,
-                        unsigned int &                        mask) const;
+      setup_constraints(
+        std::vector<types::global_dof_index> &             dof_indices,
+        const CellIterator &                               cell,
+        std::shared_ptr<const Utilities::MPI::Partitioner> partitioner,
```

Comment:
As long as we don't have a public interface accepting a `shared_ptr` or allow to access the `shared_ptr` from the outside, I don't see a reason to prefer `std::shared_ptr` over `std::unique_ptr` or `boost::optional`.

Reply:
I don't understand what you mean. I need the `partitioner` from `cuda_matrix_free.templates.h` which may or may not be initialized. If it's not (serial case) I don't use it, if it is (MPI case) then I use it.

Reply:
I just wanted to know if there is a reason to prefer `std::shared_ptr` over ` std::unique_ptr` here.

Reply:
I have switched to `unique_ptr`